### PR TITLE
Add __await__ method to Dask collections

### LINKED ISSUE #4900: Define __await__ method for collections
What should this do?

```python
async def f():
    df = dask.datasets.timeseries()
    df = await df
```

One option, would be that it is shorthand for the following:

```python
async def f():
    df = dask.datasets.timeseries()
    df = df.persist()
    await wait(df)
```

### PR DESCRIPTION
**Pull request by mrocklin**
Fixes #4900

### CODE REVIEW

**Comment by mrocklin** [comment]
@jcrist you might have thoughts here

**Comment by mrocklin** [comment]
Merging tomorrow if there are no comments here.

File: dask/base.py [hunk 1]
```diff
@@ -156,6 +156,19 @@ def compute(self, **kwargs):
         (result,) = compute(self, traverse=False, **kwargs)
         return result
 
+    def __await__(self):
+        from distributed import wait, futures_of
```
Comment (jakirkham) [inline dask/base.py]: What happens when the user doesn't have `distributed` installed?

**Comment by jcrist** [comment]
I don't think collections should be awaitable at all. Collections define operations that generally aren't scheduled at all, and the standard methods (`compute`, `persist`, etc...), along with the special distributed methods (`wait`, `as_completed`) define computations.

I could see an argument for something like the following working though:

```
# Wait until all pending futures in the awaited collection are completed
persisted = await collection.persist()
```

**Comment by mrocklin** [comment]
I also thought about having `await` only call `wait`, but I couldn't come up with a situation where I wouldn't want the `persist` call in there as well, so I thought it best to include it.

**Comment by mrocklin** [comment]
So I guess turning that around a bit, what is the value to dropping the persist call?  Are there situations where we're going to want those two actions to be separable?

**Comment by jcrist** [comment]
I think it's more of a matter of being explicit - it's not clear to me what awaiting a collection would do.

```python
# Does this compute? (my first guess, especially since await
# is a function from `Awaitable` to the awaited result)
x2 = await x
```

When awaiting a persist operation it's easy to see what's going on:

```python
# From the code alone it can be inferred that this awaits until persist is done,
# even for people not familiar with dask
x2 = await x.persist()
```

**Comment by mrocklin** [comment]
OK, I've removed the auto-persist for now.  It's easier to add it later than to remove it.

File: dask/base.py (hunk 1 above)
Comment (mrocklin) [reply dask/base.py]: The user will get an `ImportError`.  There isn't much we can do about this.  I've added a slightly more informative error message.  If you have a particular suggestion on what should happen here then please do share.

Comment (jakirkham) [reply dask/base.py]: Thanks for the fix. I think what you have pushed is good. 🙂

**Comment by jakirkham** [comment]
@jcrist, could you please give this another look?

**Comment by jcrist** [comment]
Fine by me

**Comment by jakirkham** [comment]
Thanks 🙂

**Pull request merged**